share/ipld/nmt_hasher_integration_test.go

```diff
@@ -0,0 +1,162 @@
+	}
+
+	data, err := m.Datastore.Get(ctx, ds.NewKey(key))
+	if err != nil {
+		return nil, err
+	}
+	return NewCorruptBlock(data, cid), nil
+}
+
+func (m MockBlockstore) GetSize(ctx context.Context, cid cid.Cid) (int, error) {
+	key := cid.String()
+	if m.attacking {
+		key = "corrupt" + key
+	}
+
+	return m.Datastore.GetSize(ctx, ds.NewKey(key))
+}
+
+func (m MockBlockstore) Put(ctx context.Context, block blocks.Block) error {
+	err := m.Datastore.Put(ctx, ds.NewKey(block.Cid().String()), block.RawData())
+	if err != nil {
+		return err
+	}
+
+	// create data that doesn't match the CID with arbitrary lengths between 0 and len(block.RawData())*2
+	corrupted := make([]byte, mrand.Int()%(len(block.RawData())*2))
+	mrand.Read(corrupted)
+	return m.Datastore.Put(ctx, ds.NewKey("corrupt"+block.Cid().String()), corrupted)
+}
+
+func (m MockBlockstore) PutMany(ctx context.Context, blocks []blocks.Block) error {
+	for _, b := range blocks {
+		err := m.Put(ctx, b)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m MockBlockstore) DeleteBlock(ctx context.Context, cid cid.Cid) error {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (m MockBlockstore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func (m MockBlockstore) HashOnRead(enabled bool) {
+	// TODO implement me
+	panic("implement me")
+}
+
+func MockNode(t *testing.T, net *availability_test.DagNet) (*availability_test.Node, *MockBlockstore) {
+	t.Helper()
+	dstore := dssync.MutexWrap(ds.NewMapDatastore())
+	mockBS := &MockBlockstore{
+		Datastore: dstore,
+		attacking: false,
+	}
+	provider := net.NodeWithBlockStore(dstore, mockBS)
+	provider.ShareService = service.NewShareService(provider.BlockService, full.TestAvailability(provider.BlockService))
+	return provider, mockBS
+}
+
+// TestNamespaceHasher_CorruptedData is an integration test that
+// verifies that the NamespaceHasher of a recipient of corrupted data will not panic,
+// and will throw away the corrupted data.
+func TestNamespaceHasher_CorruptedData(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	net := availability_test.NewTestDAGNet(ctx, t)
+
+	requestor := full.Node(net)
+	provider, mockBS := MockNode(t, net)
+	net.ConnectAll()
+
+	// before the provider starts attacking, we should be able to retrieve successfully
+	root := availability_test.RandFillBS(t, 16, provider.BlockService)
+	getCtx, cancelGet := context.WithTimeout(ctx, 2*time.Second)
+	t.Cleanup(cancelGet)
+	err := requestor.SharesAvailable(getCtx, root)
+	require.NoError(t, err)
+
+	// clear the storage of the requestor so that it must retrieve again, then start attacking
+	requestor.ClearStorage()
+	mockBS.attacking = true
+	getCtx, cancelGet = context.WithTimeout(ctx, 2*time.Second)
+	t.Cleanup(cancelGet)
+	err = requestor.SharesAvailable(getCtx, root)
+	require.Error(t, err, "availability vailidation failed")
```

Comment:
Are you intending to check for ErrorIs here? 

```suggestion
	require.Error(t, err, "availability validation failed")
```

Reply:
done

Reply:
https://github.com/celestiaorg/celestia-node/pull/1256/files#diff-c374312b3ea40d59f26c88cd7b963b7db87bbeeb9e169d4419a7e960cd2f7503R45

Reply:
I fixed the typo, but no its not supposed to be `ErrorIs`, because it contains a longer string and components that I don't have access to from the outside